server.go

```diff
@@ -929,6 +930,12 @@ func (s *Server) processUnaryRPC(t transport.ServerTransport, stream *transport.
 		}
 	}
 
+	var inPayload *stats.InPayload
+	if sh != nil {
+		inPayload = &stats.InPayload{
+			BeginTime: time.Now(),
```

Comment:
This time is not when the first byte is received. The bytes for this message could have been in transport's receiving queue for a while, it's just not read by gRPC.

In many cases, especially for streaming RPCs, similar to send(), this only measures time spent on decoding and decompression.